core/src/main/scala/org/apache/spark/scheduler/Task.scala

```diff
@@ -106,7 +106,21 @@ private[spark] abstract class Task[T](val stageId: Int, var partitionId: Int) ex
     if (interruptThread && taskThread != null) {
       taskThread.interrupt()
     }
-  }  
+  }
+
+  override def hashCode(): Int = {
+    31 * stageId.hashCode() + partitionId.hashCode()
+  }
+
+  def canEqual(other: Any): Boolean = other.isInstanceOf[Task[_]]
+
+  override def equals(other: Any): Boolean = other match {
+    case that: Task[_] =>
+      (that canEqual this) &&
```

Comment:
yes, in current  spark code, is the same type, but in parameter class level, I think is more reasonable to add that.